system/lib/wasmfs/memory_backend.h

```diff
@@ -58,9 +58,7 @@ class MemoryDirectory : public Directory {
     entries.push_back({name, child});
   }
 
-  std::shared_ptr<File> getChild(const std::string& name) override {
-    return nullptr;
-  }
+  std::shared_ptr<File> getChild(const std::string& name) override;
```

Comment:
This means `MemoryBackend` will always use linear lookup, right? It was returning `nullptr` so that `dcache` was used.

Reply:
Yes, that's correct. I'm generally not too worried about this because the in-memory backend is still so much faster than any persistent backend.